gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java

```diff
@@ -63,8 +63,7 @@ protected AbstractJKubeTask(Class<? extends KubernetesExtension> extensionClass)
     kubernetesExtension = getProject().getExtensions().getByType(extensionClass);
   }
 
-  @TaskAction
-  public final void runTask() {
+  protected void init() {
```

Comment:
Any reason to extract this logic into a separate method?

Reply:
I think Sun did it to make it look similar to what we have in maven plugin mojos

Reply:
I did that for readability. Then we can discuss about the method names.

Reply:
I guess, I didn't review those yet. But I don't understand the need for this (maybe in Maven is to be able to prepare the tests, but shouldn't be necessary).
I did give a quick glance on what was happening on Maven last week. What I noticed was that after that PR we would have execute, executeInternal, doExecute, etc. which would make it very hard to maintain.


In this case it makes it harder to read and it doesn't provide any added value such as reusability, etc. Condensing 3 lines into a method might not be that good ATM (maybe if the method grows in the future it will).

Reply:
> I did that for readability. Then we can discuss about the method names.

As I said, IMHO it makes it readable when the logic contains 10s of lines or can be reused. If not, it forces you to go up and down the code  to check what's happening

Reply:
Ok I moved the main method `runTask()` in the top: someone who is reading this class would see this method like a table of  content. Reader should know and understand what it is doing without having to go in the details. Then if the reader is interested to go to the details of a specific method, he can click on the method and read the content.

Reply:
the second thing is to avoid having more than 1 logic in 1 method.

Reply:
IMHO this is neat. Before, `runTask()` were like mainly about init stuff whereas now it is clearly 3 things:
- `init`
- `skipping or not`,
- `executing`

![Selection_585](https://user-images.githubusercontent.com/650571/148745427-92a6225d-9428-4cc9-bf8d-d914fd159d0c.png)

Reply:
This is also raising a question: should `init` be before or after `skip`?

Reply:
OK, in the GH interface it seemed that there were only 3 lines in the method.

we can keep it this way, but make init private.

Reply:
> This is also raising a question: should `init` be before or after `skip`?

I guess that some `canExecute` implementations depend on the init. Even your current implementation relies on having the logger initiated.